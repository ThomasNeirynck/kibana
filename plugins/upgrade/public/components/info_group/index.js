export { InfoGroup } from './info_group';
