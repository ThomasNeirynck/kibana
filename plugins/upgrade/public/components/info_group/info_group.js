import classNames from 'classnames';
import React from 'react';


export function InfoGroup({
  children,
  className,
  isCollapsed,
  onChangeCollapsed,
  title,
}) {
  const infoToggleClasses = classNames('kuiIcon', {
    'fa-chevron-circle-down': isCollapsed,
    'fa-chevron-circle-up': !isCollapsed,
  });

  return (
    <div className={ className }>
      <div className="kuiBar kuiVerticalRhythm">
        <div className="kuiBarSection">
          <h2 className='kuiTitle'>{ title }</h2>
        </div>
        <div className="kuiBarSection">
          <button className="kuiCollapseButton" onClick={ onChangeCollapsed }>
            <div className={ infoToggleClasses } />
          </button>
        </div>
      </div>

      { isCollapsed ? null : children }
    </div>
  );
}

InfoGroup.propTypes = {
  children: React.PropTypes.node,
  className: React.PropTypes.string,
  isCollapsed: React.PropTypes.bool,
  onChangeCollapsed: React.PropTypes.func,
  title: React.PropTypes.node,
};

InfoGroup.defaultProps = {
  children: null,
  className: '',
  isCollapsed: false,
  onChangeCollapsed: () => {},
  title: null,
};
