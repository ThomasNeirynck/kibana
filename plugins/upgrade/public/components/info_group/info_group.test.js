import { render, shallow } from 'enzyme';
import React from 'react';
import sinon from 'sinon';

import { InfoGroup } from './info_group';


describe('InfoGroup', () => {
  test('renders without properties', () => {
    const component = (
      <InfoGroup />
    );

    expect(render(component)).toMatchSnapshot();
  });

  test('renders the className property', () => {
    const component = (
      <InfoGroup className="specificInfoGroup" />
    );

    expect(render(component)).toMatchSnapshot();
  });

  test('renders the title property', () => {
    const component = (
      <InfoGroup title="A different title" />
    );

    expect(render(component)).toMatchSnapshot();
  });

  test('does not render its children when isCollapsed is true', () => {
    const component = (
      <InfoGroup isCollapsed={true}>
        Info Message
      </InfoGroup>
    );

    expect(render(component)).toMatchSnapshot();
  });

  test('renders its children when isCollapsed is false', () => {
    const component = (
      <InfoGroup isCollapsed={false}>
        Info Message
      </InfoGroup>
    );

    expect(render(component)).toMatchSnapshot();
  });

  test('renders its children by default', () => {
    const component = (
      <InfoGroup>
        Info message
      </InfoGroup>
    );

    expect(render(component)).toMatchSnapshot();
  });

  test('calls onChangeCollapsed when the toggle button is clicked', () => {
    const handleChangeCollapsed = sinon.stub();
    const component = (
      <InfoGroup onChangeCollapsed={handleChangeCollapsed}/>
    );

    const wrapper = shallow(component);
    wrapper.find('button').simulate('click');

    expect(handleChangeCollapsed).toHaveProperty('calledOnce', true);
  });
});
