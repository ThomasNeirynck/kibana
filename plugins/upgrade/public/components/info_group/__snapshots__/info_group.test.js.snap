// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoGroup does not render its children when isCollapsed is true 1`] = `
<div
  class=""
>
  <div
    class="kuiBar kuiVerticalRhythm"
  >
    <div
      class="kuiBarSection"
    >
      <h2
        class="kuiTitle"
      />
    </div>
    <div
      class="kuiBarSection"
    >
      <button
        class="kuiCollapseButton"
      >
        <div
          class="kuiIcon fa-chevron-circle-down"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`InfoGroup renders its children by default 1`] = `
<div
  class=""
>
  <div
    class="kuiBar kuiVerticalRhythm"
  >
    <div
      class="kuiBarSection"
    >
      <h2
        class="kuiTitle"
      />
    </div>
    <div
      class="kuiBarSection"
    >
      <button
        class="kuiCollapseButton"
      >
        <div
          class="kuiIcon fa-chevron-circle-up"
        />
      </button>
    </div>
  </div>
  Info message
</div>
`;

exports[`InfoGroup renders its children when isCollapsed is false 1`] = `
<div
  class=""
>
  <div
    class="kuiBar kuiVerticalRhythm"
  >
    <div
      class="kuiBarSection"
    >
      <h2
        class="kuiTitle"
      />
    </div>
    <div
      class="kuiBarSection"
    >
      <button
        class="kuiCollapseButton"
      >
        <div
          class="kuiIcon fa-chevron-circle-up"
        />
      </button>
    </div>
  </div>
  Info Message
</div>
`;

exports[`InfoGroup renders the className property 1`] = `
<div
  class="specificInfoGroup"
>
  <div
    class="kuiBar kuiVerticalRhythm"
  >
    <div
      class="kuiBarSection"
    >
      <h2
        class="kuiTitle"
      />
    </div>
    <div
      class="kuiBarSection"
    >
      <button
        class="kuiCollapseButton"
      >
        <div
          class="kuiIcon fa-chevron-circle-up"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`InfoGroup renders the title property 1`] = `
<div
  class=""
>
  <div
    class="kuiBar kuiVerticalRhythm"
  >
    <div
      class="kuiBarSection"
    >
      <h2
        class="kuiTitle"
      >
        A different title
      </h2>
    </div>
    <div
      class="kuiBarSection"
    >
      <button
        class="kuiCollapseButton"
      >
        <div
          class="kuiIcon fa-chevron-circle-up"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`InfoGroup renders without properties 1`] = `
<div
  class=""
>
  <div
    class="kuiBar kuiVerticalRhythm"
  >
    <div
      class="kuiBarSection"
    >
      <h2
        class="kuiTitle"
      />
    </div>
    <div
      class="kuiBarSection"
    >
      <button
        class="kuiCollapseButton"
      >
        <div
          class="kuiIcon fa-chevron-circle-up"
        />
      </button>
    </div>
  </div>
</div>
`;
